pkg/replica/revision_counter.go

```diff
@@ -92,9 +90,30 @@ func (r *Replica) initRevisionCounter() error {
 	}
 	// Don't use r.revisionCache directly
 	// r.revisionCache is an internal cache, to avoid read from disk
-	// every time when counter needs to be updated.
-	// And it's protected by revisionLock
-	r.revisionCache = counter
+	atomic.SwapInt64(&r.revisionCache, counter)
```

Comment:
```suggestion
	r.revisionCache.Swap(counter) // Or similar.
```

The Golang docs suggest replacing `SwapInt64` and similar functions with `Int64`.Swap.

https://pkg.go.dev/sync/atomic#SwapInt64

This means we need to change the type of r.revisionCounter from `int64` to `atomic.Int64`, which may have other ramifications (that are hopefully positive).

The idea seems to be that the new atomic wrappers prevent developers from accidentally performing non-atomic operations on the code elsewhere.

https://medium.com/@deckarep/the-go-1-19-atomic-wrappers-and-why-to-use-them-ae14c1177ad8

IMO, in a "post Go v1.19" world, if we are switching from mutex protection to atomic protection, we should follow this convention.

Reply:
If we make this change, other similar functions will need to change in this PR. I won't comment individually on them.